<launch>
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>

  <arg name="namespace"           default="robot1"/>
  <arg name="odom_frame_id"       default="$(arg namespace)/odom"/>
  <arg name="base_frame_id"       default="$(arg namespace)/base_footprint"/>
  <arg name="odom_topic"          default="$(arg namespace)/odom" />
  <arg name="laser_topic"         default="$(arg namespace)/scan" />
  <arg name="custom_param_file"   default="$(find turtlebot3_navigation)/param/dummy.yaml"/>
  <arg name="global_frame_id"     default="map"/>

  <node pkg="move_base" type="move_base" name="move_base_node_$(arg namespace)" respawn="false" output="screen">
    <rosparam file="$(find ros_multi)/param/$(arg namespace)_costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find ros_multi)/param/$(arg namespace)_costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find ros_multi)/param/$(arg namespace)_local_costmap_params.yaml" command="load" />
    <rosparam file="$(find ros_multi)/param/$(arg namespace)_global_costmap_params.yaml" command="load" />
    <rosparam file="$(find turtlebot3_navigation)/param/dwa_local_planner_params_burger.yaml" command="load" />
    <rosparam file="$(find turtlebot3_navigation)/param/move_base_params.yaml" command="load" />

    <param name="base_local_planner"                value="dwa_local_planner/DWAPlannerROS" />
    <param name="global_costmap/global_frame"       value="$(arg global_frame_id)"/>
    <param name="global_costmap/robot_base_frame"   value="$(arg base_frame_id)"/>
    <param name="local_costmap/global_frame"        value="$(arg odom_frame_id)"/>
    <param name="local_costmap/robot_base_frame"    value="$(arg base_frame_id)"/>
    <param name="DWAPlannerROS/global_frame_id"     value="$(arg odom_frame_id)"/>

    <remap from="cmd_vel" to="/$(arg namespace)/cmd_vel"/>
    <remap from="odom"    to="/$(arg odom_topic)"/>
    <remap from="scan"    to="/$(arg laser_topic)"/>
    <remap from="map"     to="/map" />

    <remap from="/move_base_simple/goal"                                           to="/$(arg namespace)/move_base_simple/goal" />
    <remap from="/move_base/TebLocalPlannerROS/global_plan"                        to="/$(arg namespace)/move_base/TebLocalPlannerROS/global_plan" />
    <remap from="/move_base/TebLocalPlannerROS/local_plan"                         to="/$(arg namespace)/move_base/TebLocalPlannerROS/local_plan" />
    <remap from="/move_base/TebLocalPlannerROS/teb_markers"                        to="/$(arg namespace)/move_base/TebLocalPlannerROS/teb_markers" />
    <remap from="/move_base/TebLocalPlannerROS/teb_markers_array"                  to="/$(arg namespace)/move_base/TebLocalPlannerROS/teb_markers_array" />
    <remap from="/move_base/TebLocalPlannerROS/teb_poses"                          to="/$(arg namespace)/move_base/TebLocalPlannerROS/teb_poses" />
    <remap from="/move_base/global_costmap/costmap"                                to="/$(arg namespace)/move_base/global_costmap/costmap" />
    <remap from="/move_base/global_costmap/costmap_updates"                        to="/$(arg namespace)/move_base/global_costmap/costmap_updates" />
    <remap from="/move_base/local_costmap/costmap"                                 to="/$(arg namespace)/move_base/local_costmap/costmap" />
    <remap from="/move_base/local_costmap/costmap_updates"                         to="/$(arg namespace)/move_base/local_costmap/costmap_updates" />
    <remap from="/move_base/local_costmap/footprint"                               to="/$(arg namespace)/move_base/local_costmap/footprint" />

    <remap from="/move_base/GlobalPlanner/parameter_descriptions"                  to="/$(arg namespace)/move_base/GlobalPlanner/parameter_descriptions" />
    <remap from="/move_base/GlobalPlanner/parameter_updates"                       to="/$(arg namespace)/move_base/GlobalPlanner/parameter_updates" />
    <remap from="/move_base/GlobalPlanner/plan"                                    to="/$(arg namespace)/move_base/GlobalPlanner/plan" />
    <remap from="/move_base/GlobalPlanner/potential"                               to="/$(arg namespace)/move_base/GlobalPlanner/potential" />
    <remap from="/move_base/TebLocalPlannerROS/obstacles"                          to="/$(arg namespace)/move_base/TebLocalPlannerROS/obstacles" />
    <remap from="/move_base/TebLocalPlannerROS/parameter_descriptions"             to="/$(arg namespace)/move_base/TebLocalPlannerROS/parameter_descriptions" />
    <remap from="/move_base/TebLocalPlannerROS/parameter_updates"                  to="/$(arg namespace)/move_base/TebLocalPlannerROS/parameter_updates" />
    <remap from="/move_base/cancel"                                                to="/$(arg namespace)/move_base/cancel" />
    <remap from="/move_base/current_goal"                                          to="/$(arg namespace)/move_base/current_goal" />
    <remap from="/move_base/feedback"                                              to="/$(arg namespace)/move_base/feedback" />
    <remap from="/move_base/global_costmap/footprint"                              to="/$(arg namespace)/move_base/global_costmap/footprint" />
    <remap from="/move_base/global_costmap/inflation_layer/parameter_descriptions" to="/$(arg namespace)/move_base/global_costmap/inflation_layer/parameter_descriptions" />
    <remap from="/move_base/global_costmap/inflation_layer/parameter_updates"      to="/$(arg namespace)/move_base/global_costmap/inflation_layer/parameter_updates" />
    <remap from="/move_base/global_costmap/obstacle_layer/clearing_endpoints"      to="/$(arg namespace)/move_base/global_costmap/obstacle_layer/clearing_endpoints" />
    <remap from="/move_base/global_costmap/obstacle_layer/parameter_descriptions"  to="/$(arg namespace)/move_base/global_costmap/obstacle_layer/parameter_descriptions" />
    <remap from="/move_base/global_costmap/obstacle_layer/parameter_updates"       to="/$(arg namespace)/move_base/global_costmap/obstacle_layer/parameter_updates" />
    <remap from="/move_base/global_costmap/parameter_descriptions"                 to="/$(arg namespace)/move_base/global_costmap/parameter_descriptions" />
    <remap from="/move_base/global_costmap/parameter_updates"                      to="/$(arg namespace)/move_base/global_costmap/parameter_updates" />
    <remap from="/move_base/global_costmap/static_layer/parameter_descriptions"    to="/$(arg namespace)/move_base/global_costmap/static_layer/parameter_descriptions" />
    <remap from="/move_base/global_costmap/static_layer/parameter_updates"         to="/$(arg namespace)/move_base/global_costmap/static_layer/parameter_updates" />
    <remap from="/move_base/goal"                                                  to="/$(arg namespace)/move_base/goal" />
    <remap from="/move_base/local_costmap/obstacle_layer/parameter_descriptions"   to="/$(arg namespace)/move_base/local_costmap/obstacle_layer/parameter_descriptions" />
    <remap from="/move_base/local_costmap/obstacle_layer/parameter_updates"        to="/$(arg namespace)/move_base/local_costmap/obstacle_layer/parameter_updates" />
    <remap from="/move_base/local_costmap/parameter_descriptions"                  to="/$(arg namespace)/move_base/local_costmap/parameter_descriptions" />
    <remap from="/move_base/local_costmap/parameter_updates"                       to="/$(arg namespace)/move_base/local_costmap/parameter_updates" />
    <remap from="/move_base/local_costmap/static_layer/parameter_descriptions"     to="/$(arg namespace)/move_base/local_costmap/static_layer/parameter_descriptions" />
    <remap from="/move_base/local_costmap/static_layer/parameter_updates"          to="/$(arg namespace)/move_base/local_costmap/static_layer/parameter_updates" />
    <remap from="/move_base/parameter_descriptions"                                to="/$(arg namespace)/move_base/parameter_descriptions" />
    <remap from="/move_base/parameter_updates"                                     to="/$(arg namespace)/move_base/parameter_updates" />
    <remap from="/move_base/result"                                                to="/$(arg namespace)/move_base/result" />
    <remap from="/move_base/status"                                                to="/$(arg namespace)/move_base/status" />
    <remap from="/move_base_simple/goal"                                           to="/$(arg namespace)/move_base_simple/goal" />
  </node>
</launch>
